functions/src/glean.ts

```diff
@@ -0,0 +1,276 @@
+  const { default: Glean } = await import("@mozilla/glean/node");
+  const rallyPings = await initializeGlean(Glean, studyID, rallyID); // study ID is app ID
+
+  const unenrollmentMetrics = await import("./generated/unenrollment.js");
+  unenrollmentMetrics.studyId.set(studyID);
+  rallyPings.studyUnenrollment.submit();
+
+  Glean.setUploadEnabled(false); // automatically trigger a deletion-request ping
+  await Glean.shutdown();
+}
+
+/*
+ * Helper function for initializing Glean
+ */
+async function initializeGlean(
+  Glean: any,
+  appID: string,
+  rallyID: string,
+  uploadEnabled: boolean = true
+) {
+  Glean.setDebugViewTag(GLEAN_DEBUG_VIEW_TAG);
+  Glean.setLogPings(true);
+  Glean.initialize(appID, true, {
+    appDisplayVersion: GLEAN_APP_DISPLAY_VERSION,
+    plugins: [new PingEncryptionPlugin(GLEAN_ENCRYPTION_JWK)],
+    httpClient: new CustomPingUploader(),
+  });
+
+  Glean.setUploadEnabled(uploadEnabled);
+  const rallyMetrics = await import("./generated/rally.js");
+  rallyMetrics.id.set(rallyID);
+  const rallyPings = await import("./generated/pings.js");
+  return rallyPings; // Always need rallyPings, so return it to initializer
+}
+
+/*
+ * Helper function for setting user metrics
+ * from demographic data (mapping)
+ */
+function setUserMetrics(userMetrics: any, data: any) {
+  if ("age" in data) {
+    userMetrics.age[`band_${data["age"]}`].set(true);
+  }
+
+  if ("gender" in data) {
+    userMetrics.gender[data["gender"]].set(true);
+  }
+
+  if ("hispanicLatinxSpanishOrigin" in data) {
+    const label =
+      data["hispanicLatinxSpanishOrigin"] === "other"
+        ? "other"
+        : "hispanic_latinx_spanish";
+    userMetrics.origin[label].set(true);
+  }
+
+  if ("race" in data) {
+    for (const raceLabel of data["race"]) {
+      const label =
+        raceLabel === "american_indian_or_alaska_native"
+          ? "am_indian_or_alaska_native"
+          : raceLabel;
+      userMetrics.races[label].set(true);
+    }
+  }
+
+  if ("school" in data) {
+    const KEY_FIXUP: any = {
+      high_school_graduate_or_equivalent: "high_school_grad_or_eq",
+      some_college_but_no_degree_or_in_progress: "college_degree_in_progress",
+    };
+
+    const originalLabel = data["school"];
+    const label =
+      originalLabel in KEY_FIXUP ? KEY_FIXUP[originalLabel] : originalLabel;
+    userMetrics.school[label].set(true);
+  }
+
+  if ("exactIncome" in data) {
+    userMetrics.exactIncome.set(data["exactIncome"]);
+  }
+
+  if ("zipcode" in data) {
+    userMetrics.zipcode.set(data["zipcode"]);
+  }
+}
+
+/**
+ * Custom Ping Uploader for Glean
+ * TODO: replace direct POST request with Google Cloud Task
+ */
+class CustomPingUploader extends Uploader {
+  async post(
+    url: string,
+    body: string | Uint8Array,
+    headers: any
+  ): Promise<UploadResult> {
+    return await axios
+      .post(url, body, { headers: headers })
+      .then(function (response) {
+        // console.log(response);
```

Comment:
You should consider using the [cloud functions logger](https://firebase.google.com/docs/functions/writing-and-viewing-logs), I think it would be fine to have some informational logs that show up in the cloud logging tools.